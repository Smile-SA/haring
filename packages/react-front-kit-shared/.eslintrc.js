module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
