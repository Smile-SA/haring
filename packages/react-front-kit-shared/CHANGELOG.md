# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
