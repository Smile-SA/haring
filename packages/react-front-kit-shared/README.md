# React Front Kit shared

Shared components and functions.

## Documentation

You can access the documentation here: https://smile-sa.github.io/react-front-kit-doc

## Prerequisite

You must have React 18 or later installed on your project.

## Installation

Install `@smile/react-front-kit-shared` and peer dependencies with:

```bash
npm i @smile/react-front-kit-shared @mantine/core @mantine/hooks @phosphor-icons/react
```

## Styles

Import the styles in your app with:

```js
import '@mantine/core/styles.css';
import '@smile/react-front-kit-shared/style.css';
```

## License

LGPL-3.0
