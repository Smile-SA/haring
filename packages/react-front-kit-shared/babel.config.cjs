const { babelConfig } = require('test');

module.exports = babelConfig;
