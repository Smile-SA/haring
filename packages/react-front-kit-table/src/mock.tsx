/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
