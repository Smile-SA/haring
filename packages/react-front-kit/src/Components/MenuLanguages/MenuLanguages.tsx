'use client';

import type { IButtonListOrDropdownProps } from '../ButtonListOrDropdown/ButtonListOrDropdown';
import type { IOption } from '@smile/react-front-kit-shared';
import type { ReactElement } from 'react';

import { ButtonListOrDropdown } from '../ButtonListOrDropdown/ButtonListOrDropdown';

import classes from './MenuLanguages.module.css';

export interface ILang extends IOption<string> {
  country?: string;
}

export interface IMenuLanguagesProps
  extends Omit<IButtonListOrDropdownProps, 'items'> {
  languages: ILang[];
  squareFormat?: boolean;
}

export function MenuLanguages(props: IMenuLanguagesProps): ReactElement {
  const {
    languages,
    squareFormat = false,
    ...ButtonListOrDropdownProps
  } = props;

  const items = languages.map((language) => {
    const classNames = ['fi', `fi-${language.country?.toLowerCase()}`];
    if (squareFormat) {
      classNames.push('fis');
    }
    if (language.label) {
      classNames.push(classes.flag as string);
    }
    return {
      content: (
        <>
          {Boolean(language.country) && <i className={classNames.join(' ')} />}
          {language.label}
        </>
      ),
      label: language.label,
      value: language.value,
    };
  });

  return (
    <div>
      <ButtonListOrDropdown items={items} {...ButtonListOrDropdownProps} />
    </div>
  );
}
