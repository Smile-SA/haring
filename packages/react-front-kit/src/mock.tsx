/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
