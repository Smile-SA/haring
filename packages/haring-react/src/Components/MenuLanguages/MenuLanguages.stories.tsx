import type { Meta, StoryObj } from '@storybook/react';

import { MenuLanguages as Cmp } from './MenuLanguages';
import { languages } from './MenuLanguages.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/MenuLanguages',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const MenuLanguages: IStory = {
  args: {
    defaultCurrent: 'FR',
    languages,
    maxButtonItems: 0,
    squareFormat: false,
  },
};
