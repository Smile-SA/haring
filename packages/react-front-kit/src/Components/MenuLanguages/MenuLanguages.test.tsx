import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { MenuLanguages } from './MenuLanguages';
import { languages } from './MenuLanguages.mock';

describe('MenuLanguages', () => {
  beforeEach(() => {
    // Prevent mantine random ID
    Math.random = () => 0.42;
  });
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <MenuLanguages defaultCurrent="FR" languages={languages} />,
    );
    expect(container).toMatchSnapshot();
  });
});
