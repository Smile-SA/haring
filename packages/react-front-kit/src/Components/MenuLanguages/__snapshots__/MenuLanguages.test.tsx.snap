// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLanguages matches snapshot 1`] = `
<div>
  <div>
    <button
      aria-controls="mantine-f4bipx4bi-dropdown"
      aria-expanded="false"
      aria-haspopup="menu"
      class="mantine-UnstyledButton-root mantine-Button-root mantine-1w0871i"
      data-button="true"
      id="mantine-f4bipx4bi-target"
      type="button"
    >
      <div
        class="mantine-1wpc1xj mantine-Button-inner"
      >
        <span
          class="mantine-1ryt1ht mantine-Button-label"
        >
          <i
            class="fi fi-fr mantine-7umzq0"
          />
          Fran√ßais
        </span>
      </div>
    </button>
  </div>
</div>
`;
